feat: Update version of OneTrust CMP provider to 202210.1.0.0

alexzchut opened the pull request:
## Description
Update version of OneTrust CMP provider to 202210.1.0.0
https://developer.onetrust.com/onetrust/docs/changelog-ios-tvos#20221010

[Monday ticket] https://applicaster.monday.com/boards/2742545177/pulses/3540944306

`Please describe PR here`

Includes changes from https://github.com/applicaster/Zapp-Frameworks/pull/1237

[Monday ticket](https://applicaster.monday.com/boards/1615228456/pulses/3443595679)

`If merged this will fix the issues of two native screens opening in AndroidTV and tvOS platforms and also fix the issue when opening the plugin from a settings screen`


<details>
  <summary>Platforms Affected</summary>
  
- [ ] Android
- [ ] Android TV
- [ ] Apple TV (TVOS)
- [ ] Fire TV (FireOS)
- [ ] iOS
- [ ] LG TV (WebOS)
- [ ] Samsung TV (Tizen)
- [ ] Web
</details>

## Known or Pending Issues

Native platforms still run onetrust web check, code separation is flawed. (slows down init)

LG has focus issues on preferences intermediate screen

## Screenshots

`If there is some kind of ui change please share screenshots`

## Version Info

`Does this update require a certain version of quickbrick, or does it rely on another pr to work?`

ranmeirman commented on plugins/applicaster-cmp-onetrust/manifests/manifest.config.js:
```diff
@@ -200,15 +200,11 @@ const custom_configuration_fields_shared = [
+    type: "text",
+    label: "SDK URL or version",
+    initial_value: "",
+    tooltip_text: "Onetrust SDK path or version to use with the app",
```
why should it be path? let's only use version, it is confusing.

alexzchut commented:
@ranmeirman I updated the label, please review again

ranmeirman commented on plugins/applicaster-cmp-onetrust/manifests/manifest.config.js:
```diff
@@ -200,15 +200,11 @@ const custom_configuration_fields_shared = [
-  }
+    type: "text",
+    label: "SDK version",
+    initial_value: "",
```
We don't want a default value? Also let's add, "OneTrust requires that the version set in this field to match the SDK version published on the OneTrust portal. In case the versions will not match the OneTrust screen will not work properly".

alexzchut replied:
@ranmeirman 
There is a fallback to the version specified in plugin (on iOS/tvOS it is on podspec) so if no value added here, it will take the predefined tested version. Basically we can put the same predefined version here to let the user know what is it and let him change if needed.
I will change the description

alexzchut replied:
Having the version specified also here as default value will actually require a need of updating it in several places and in addition it can be different for each platform

alexzchut commented on plugins/applicaster-cmp-onetrust/manifests/manifest.config.js:
```diff
@@ -200,15 +200,11 @@ const custom_configuration_fields_shared = [
+    type: "text",
+    label: "SDK version",
+    initial_value: external_sdk_version[platform],
+    tooltip_text: "OneTrust requires that the version set in this field to match the SDK version published on the OneTrust portal. In case the versions will not match the OneTrust screen will not work properly",
```
@ranmeirman I added the initial value based on the platform so it is defined below for each one like we have on sdks

alexzchut commented on plugins/applicaster-cmp-onetrust/manifests/manifest.config.js:
```diff
@@ -240,6 +236,17 @@ const min_zapp_sdk = {
+  samsung_tv: external_sdk_version_default,
+  lg_tv: external_sdk_version_default
+};
+
```
here

alexzchut commented:
@ranmeirman I updated the tooltip and also added the default values per platform that will be set on plugin, we will keep in mind to update these values if needed, FYI @inckie

ranmeirman reviewed (approved):
From the manifest perspective it is approved, the rest should be reviewed by someone that knows more the details.

mmatsuk-sc commented:
Native part is finished, i guess. Need to review it with Alex. I also think configuration property `show_intro_screen` could be removed from the native side config. It will be used for the web platforms only
